protocol/v2/ssv/validator/committee.go

```diff
@@ -101,35 +101,45 @@ func (c *Committee) StartDuty(logger *zap.Logger, duty *spectypes.CommitteeDuty)
 	}
 
 	slashableValidators := make([]spectypes.ShareValidatorPK, 0, len(duty.BeaconDuties))
-	validatorShares := make(map[phase0.ValidatorIndex]*spectypes.Share, len(duty.BeaconDuties))
-	toRemove := make([]int, 0)
-	// Remove beacon duties that don't have a share
-	for i, bd := range duty.BeaconDuties {
-		share, ok := c.Shares[bd.ValidatorIndex]
-		if !ok {
-			toRemove = append(toRemove, i)
-			continue
-		}
-		if bd.Type == spectypes.BNRoleAttester {
-			slashableValidators = append(slashableValidators, share.SharePubKey)
-		}
-		validatorShares[bd.ValidatorIndex] = share
-	}
-	// Remove beacon duties that don't have a share
-	if len(toRemove) > 0 {
-		newDuties, err := removeIndices(duty.BeaconDuties, toRemove)
-		if err != nil {
-			logger.Warn("could not remove beacon duties", zap.Error(err), zap.Ints("indices", toRemove))
-		} else {
-			duty.BeaconDuties = newDuties
-		}
-	}
 
-	if len(duty.BeaconDuties) == 0 {
-		return errors.New("CommitteeDuty has no valid beacon duties")
+	// TODO bring this back when https://github.com/ssvlabs/ssv-spec/pull/467 is merged and spec is aligned
+	//validatorShares := make(map[phase0.ValidatorIndex]*spectypes.Share, len(duty.BeaconDuties))
+	//toRemove := make([]int, 0)
+	//// Remove beacon duties that don't have a share
+	//for _, bd := range duty.BeaconDuties {
+	//	share, ok := c.Shares[bd.ValidatorIndex]
+	//	if !ok {
+	//		toRemove = append(toRemove, i)
+	//		continue
+	//	}
+	//	if bd.Type == spectypes.BNRoleAttester {
+	//		slashableValidators = append(slashableValidators, share.SharePubKey)
+	//	}
+	//	validatorShares[bd.ValidatorIndex] = share
+	//}
+	//// Remove beacon duties that don't have a share
+	//if len(toRemove) > 0 {
+	//	newDuties, err := removeIndices(duty.BeaconDuties, toRemove)
+	//	if err != nil {
+	//		logger.Warn("could not remove beacon duties", zap.Error(err), zap.Ints("indices", toRemove))
+	//	} else {
+	//		duty.BeaconDuties = newDuties
+	//	}
+	//}
+	//
+	//if len(duty.BeaconDuties) == 0 {
+	//	return errors.New("CommitteeDuty has no valid beacon duties")
+	//}
+	//
+	//r := c.CreateRunnerFn(duty.Slot, validatorShares, slashableValidators)
+
+	// TODO REMOVE this after https://github.com/ssvlabs/ssv-spec/pull/467 is merged and we are aligned to the spec -->
+	var sharesCopy = make(map[phase0.ValidatorIndex]*spectypes.Share, len(c.Shares))
+	for k, v := range c.Shares {
+		sharesCopy[k] = v
 	}
-
-	r := c.CreateRunnerFn(duty.Slot, validatorShares, slashableValidators)
+	r := c.CreateRunnerFn(duty.Slot, sharesCopy, slashableValidators)
+	// <--
```

Comment:
you still have to filter `slashableValidators`, according to these who have attestations only. right now you are passing an empty slice.

Reply:
Done. Please check